### DISCUSSION TITLE
Is it a bug. when hit the "enter" key , got a step with two paragraph

### DISCUSSION THREAD

**Discussion by 4-1-1** [Milkdown/milkdown#668] [Q&A]
Hit  "enter"  at an empty line to create a new paragraph.
But the tr step will carry two  paragraph

<img width="1100" alt="image" src="https://user-images.githubusercontent.com/4401981/184087778-a92c12df-0090-4cc9-b4d9-0f938614b4b9.png">

**Comment by Saul-Mirone** [comment]
That's by design of prosemirror.
https://codesandbox.io/s/heuristic-rhodes-fqxloy?file=/src/index.ts

**Comment by 4-1-1** [comment]
It's quite wired.
I need to calc the content length change by the step. to get the total length and avoid the serializing.

A lot of "sigh"!
